src/cachingreader.h

```diff
@@ -34,6 +35,8 @@ typedef struct Hint {
     int priority;
 } Hint;
 
+typedef QVarLengthArray<Hint, 512> HintVector;
```

Comment:
Are you sure, that this is actually an improvement? 
QVarLengthArray allocates the first 512 hints on the stack, but since this is a member variable, there is probably no stack, because EngineBuffer is instantiated with new. I did not look at the Qt code, but I think there should be a small performance penalty caused by the heap and stack support compared to QVector. 

On the other hand QVector has an improved reallocation in case 512 is not enough. 

But what I can remember 512 was a lot more then enough, when I have introduced it. 

So without any test results, I am convinced the original version is faster by a very small amount.